src/Dict/Dict.res

```diff
@@ -23,6 +23,13 @@ export fromPairs = dict => Js.Dict.fromArray(dict)
 %comment("Merges two provided objects.")
 export merge = (fst, snd) => assign(makeEmpty(), fst, snd)
 
+%comment("Adds a property. Equivalent to merging with `{key: value}`")
+export set = (dict, key, value) => {
+  let pairs = toPairs(dict)
+  let appended = Array.append(pairs, (key, value))
+  fromPairs(appended)
+}
```

Comment:
looks good, although, I'd use this solution (I believe there's no need to iterate):

```rescript
export set = (dict, key, value) => {
  let obj = assign(makeEmpty(), dict)
  Js.Dict.set(obj, key, value)
  obj
}
```

Reply:
Yes, that's really nice. I believe you mean `merge` instead of  `assign`?

Reply:
good point, both are ok 💯